### ISSUE TITLE
Fix loading of activesupport for v7.1.x

### ISSUE THREAD

**Issue by abrom** [theodorton/yard-activerecord#36]
The loading of `active_support/core_ext/array` in `yard-activerecord/validations/validates_handler.rb` breaks when using ActiveSupport 7.1.x because it has a prerequisite dependency that isn't explicitly loaded:

```
[error]: Invalid configuration file, using default options.
[error]: NoMethodError: undefined method `deprecator' for module ActiveSupport
[error]: Stack trace:
	/<path to gems>/activesupport-7.1.3.4/lib/active_support/core_ext/array/conversions.rb:108:in `<class:Array>'
	/<path to gems>/activesupport-7.1.3.4/lib/active_support/core_ext/array/conversions.rb:8:in `<top (required)>'
	<internal:/<path to gems>/core_ext/kernel_require.rb>:37:in `require'
	<internal:/<path to gems>/core_ext/kernel_require.rb>:37:in `require'
	/<path to gems>/activesupport-7.1.3.4/lib/active_support/core_ext/array.rb:5:in `<top (required)>'
	<<path to gems>/core_ext/kernel_require.rb>:37:in `require'
```

It looks like there are two solutions.

1. Add `require 'active_support/deprecator'` before trying to require `active_support/core_ext/array`
2. Add `require 'active_support'` (possibly to `yard-activerecord.rb`)

I would suggest option 2 as this is per the recommendation in the ActiveSupport documentation. See https://guides.rubyonrails.org/active_support_core_extensions.html#cherry-picking-a-definition

Option 1 would likely cause problems for anyone using this gem that doesn't have a version of ActiveSupport that includes "deprecator" support.

Thoughts?

I'm happy to put together a PR 😄

**Comment by abrom** [comment]
Sorry.. ignore me. I see you've gone with option 1. per https://github.com/theodorton/yard-activerecord/commit/c5ce8a07979ee86e07933cdbaf10c1f56482fcf2 just not released as a gem yet 😄 

I would still suggest that switching to the "supported" method of cherry picking core_ext deps would be preferential

**Comment by theodorton** [comment]
Hey @abrom 👋🏻 

I agree the current loading isn't idiomatic. I made a PR with option 2 - https://github.com/theodorton/yard-activerecord/pull/37

Could you try loading that branch in your project and see if it works?

```ruby
gem 'yard-activerecord', github: 'theodorton/yard-activerecord', branch: 'activesupport-deps'
```

**Comment by abrom** [comment]
yes, looks good to me 👍

**Comment by theodorton** [comment]
Resolved in #37 and released in `v0.0.17` - thanks for the report :)

**Comment by abrom** [comment]
nice one, thanks @theodorton 😄
